### ISSUE TITLE
ingress pod failing readiness probe on AKS k8s v1.23.3 when using helm chart v1.5.1

### ISSUE THREAD

**Issue by severity1** [Azure/application-gateway-kubernetes-ingress#1385]
**Describe the bug**
Ingress pod runs but fails readiness probe.

**To Reproduce**
Install ingress-azure using helm-chart version 1.5.1 on aks using kubernetes version 1.23.3 and it will fail readiness probe.

**Ingress Controller details**
* Output of `kubectl describe pod <ingress controller`> . The <ingress controller> pod name can be obtained by running `helm list`. 

```
Name:         ingress-azure-65d474d47b-qxlr6
Namespace:    agic
Priority:     0
Node:         aks-operations-34603714-vmss000000/10.128.33.165
Start Time:   Thu, 28 Apr 2022 03:29:28 +0000
Labels:       aadpodidbinding=ingress-azure
              app=ingress-azure
              pod-template-hash=65d474d47b
              release=ingress-azure
Annotations:  checksum/config: 41f30e3bb2a61517fa08bff71484e697c0d0ca3952c57c183e896237e3bc8eaa
              prometheus.io/port: 8123
              prometheus.io/scrape: true
Status:       Running
IP:           10.128.33.192
IPs:
  IP:           10.128.33.192
Controlled By:  ReplicaSet/ingress-azure-65d474d47b
Containers:
  ingress-azure:
    Container ID:   containerd://e4041c468432d4c34ecac7648fcfb831f3f09f63472fa11db25b70c166044f66
    Image:          mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:1.4.0
    Image ID:       mcr.microsoft.com/azure-application-gateway/kubernetes-ingress@sha256:42fcfda70020e91668047b4b975410037b4e7cfa9378cbec7b81c55f0ce5aa3e
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Thu, 28 Apr 2022 03:29:29 +0000
    Ready:          False
    Restart Count:  0
    Liveness:       http-get http://:8123/health/alive delay=15s timeout=1s period=20s #success=1 #failure=3
    Readiness:      http-get http://:8123/health/ready delay=5s timeout=1s period=10s #success=1 #failure=3
    Environment Variables from:
      ingress-azure  ConfigMap  Optional: false
    Environment:
      AZURE_CLOUD_PROVIDER_LOCATION:  /etc/appgw/azure.json
      AGIC_POD_NAME:                  ingress-azure-65d474d47b-qxlr6 (v1:metadata.name)
      AGIC_POD_NAMESPACE:             agic (v1:metadata.namespace)
    Mounts:
      /etc/appgw/ from azure (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-bdlf5 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  azure:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/kubernetes/
    HostPathType:  Directory
  kube-api-access-bdlf5:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
                             workload=ops:NoSchedule
Events:
  Type     Reason     Age                   From               Message
  ----     ------     ----                  ----               -------
  Normal   Scheduled  5m56s                 default-scheduler  Successfully assigned agic/ingress-azure-65d474d47b-qxlr6 to aks-operations-34603714-vmss000000
  Normal   Pulling    5m56s                 kubelet            Pulling image "mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:1.4.0"
  Normal   Pulled     5m56s                 kubelet            Successfully pulled image "mcr.microsoft.com/azure-application-gateway/kubernetes-ingress:1.4.0" in 210.720496ms
  Normal   Created    5m56s                 kubelet            Created container ingress-azure
  Normal   Started    5m55s                 kubelet            Started container ingress-azure
  Warning  Unhealthy  55s (x34 over 5m45s)  kubelet            Readiness probe failed: Get "http://10.128.33.192:8123/health/ready": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```
* Output of `kubectl logs <ingress controller>. 
```
I0428 03:29:29.035335       1 utils.go:115] Using verbosity level 3 from environment variable APPGW_VERBOSITY_LEVEL
I0428 03:29:29.074031       1 environment.go:248] KUBERNETES_WATCHNAMESPACE is not set. Watching all available namespaces.
I0428 03:29:29.074051       1 main.go:117] Using User Agent Suffix='ingress-azure-65d474d47b-qxlr6' when communicating with ARM
I0428 03:29:29.074124       1 main.go:136] Appication Gateway Details: Subscription="bb9d781e-1b25-4e46-a21c-e2dc748fc138" Resource Group="sharedtest1-nodes-rg" Name="agic-appgw"
I0428 03:29:29.074132       1 auth.go:46] Creating authorizer from Azure Managed Service Identity
I0428 03:29:29.074265       1 httpserver.go:57] Starting API Server on :8123
I0428 03:29:39.811606       1 main.go:183] Ingress Controller will observe all namespaces.
I0428 03:29:39.928004       1 context.go:138] k8s context run started
I0428 03:29:39.928038       1 context.go:188] Waiting for initial cache sync
E0428 03:29:39.932510       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:29:39.933886       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:29:42.903872       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:29:48.567700       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:29:55.781229       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:30:18.801877       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:30:49.737275       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:31:46.937868       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:32:24.614400       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:33:07.407482       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:33:47.783955       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:34:40.225525       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:35:27.267817       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:35:58.422629       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
E0428 03:36:40.521406       1 reflector.go:178] pkg/mod/k8s.io/client-go@v0.0.0-20200326020446-6240434e1ad6/tools/cache/reflector.go:125: Failed to list *v1beta1.Ingress: the server could not find the requested resource (get ingresses.extensions)
```
* Any Azure support tickets associated with this issue.

**Comment by severity1** [comment]
curious as to why it fails when trying to list the beta ingress version when this should still work if beta version is missing. not sure if it will help to say that this is a new cluster so beta ingress api version does not exist on this cluster. See my api versions below.

```
aadpodidentity.k8s.io/v1
admissionregistration.k8s.io/v1
apiextensions.k8s.io/v1
apiregistration.k8s.io/v1
appgw.ingress.k8s.io/v1
apps/v1
authentication.k8s.io/v1
authorization.k8s.io/v1
autoscaling/v1
autoscaling/v2
autoscaling/v2beta1
autoscaling/v2beta2
batch/v1
batch/v1beta1
certificates.k8s.io/v1
coordination.k8s.io/v1
discovery.k8s.io/v1
discovery.k8s.io/v1beta1
events.k8s.io/v1
events.k8s.io/v1beta1
flowcontrol.apiserver.k8s.io/v1beta1
flowcontrol.apiserver.k8s.io/v1beta2
metrics.k8s.io/v1beta1
monitoring.coreos.com/v1
monitoring.coreos.com/v1alpha1
networking.k8s.io/v1
node.k8s.io/v1
node.k8s.io/v1beta1
policy/v1
policy/v1beta1
rbac.authorization.k8s.io/v1
scheduling.k8s.io/v1
snapshot.storage.k8s.io/v1
snapshot.storage.k8s.io/v1beta1
spv.no/v1
spv.no/v1alpha1
spv.no/v2alpha1
spv.no/v2beta1
storage.k8s.io/v1
storage.k8s.io/v1beta1
v1
```

**Comment by severity1** [comment]
This is fixed, apparently we were using the 1.4.0 image.
